lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -136,12 +143,18 @@ extension DomElementExtension on DomElement {
   external /* List<DomElement> */ List<Object?> get children;
   external String get id;
   external set id(String id);
+  external String? get outerHTML;
+  external set spellcheck(bool? value);
+  external String get tagName;
   external DomCSSStyleDeclaration get style;
   external void append(DomNode node);
   external String? getAttribute(String attributeName);
   external void prepend(DomNode node);
   external DomElement? querySelector(String selectors);
+  external /* List<DomNode> */ List<Object?> querySelectorAll(String selectors);
```

Comment:
Is it possible to reserve the name `querySelectorAll` for a wrapper with a cast to `List<DomNode>` and use a different name for the raw method?

Reply:
We could. We could actually add cast wrappers to all of these lists. If they aren't used in inner loops it's probably okay. Thoughts?

Reply:
I just went ahead and made all the list interfaces cast lists. Because the lists are lazy, the overhead should be quite minimal. Let me know if you prefer to have the raw calls around otherwise we can remove them if they ever show up in a profile.

Reply:
Actually, I went ahead and made these a `List` for now, but we will probably want to change these in the future to reflect the actual typed DOM list returned by these APIs. It shouldn't be hard to do that at the very end though.